File: frontend/src/apis/message/ListMessageBlocks.ts
import type { APIWithSessionMessage } from '@/system/api/api-message'
import type { MessageBlockEntry } from '@/objects/message/response/MessageBlockEntry'

/** 查询当前会话消息屏蔽列表；无请求体，输出被屏蔽用户条目数组。 */
export class ListMessageBlocks implements APIWithSessionMessage<MessageBlockEntry[]> {
  declare readonly responseType?: MessageBlockEntry[]
  readonly method = 'GET'
  readonly apiPath = 'messages/blocks'

  body(): undefined {
    return undefined
  }
}

File: backend/src/main/scala/domains/message/http/api/ListMessageBlocks.scala
package domains.message.http.api

import cats.effect.IO
import domains.auth.http.AuthenticatedApi
import domains.auth.objects.AuthUser
import domains.message.http.codec.MessageHttpCodecs.given
import domains.message.objects.response.MessageBlockEntry
import domains.message.table.message.MessageBlockTable
import io.circe.Encoder
import org.http4s.{Method, Request, Status}
import shared.http.{ApiPath, PathParams}

import java.sql.Connection

object ListMessageBlocks extends AuthenticatedApi[Unit, List[MessageBlockEntry]]:

  override val method: Method = Method.GET
  override val path: ApiPath = ApiPath("/api/messages/blocks")
  override val successStatus: Status = Status.Ok
  override protected val outputEncoder: Encoder[List[MessageBlockEntry]] = Encoder.encodeList[MessageBlockEntry]

  override def decode(request: Request[IO], pathParams: PathParams): IO[Unit] =
    val _ = (request, pathParams)
    IO.unit

  override def plan(connection: Connection, actor: AuthUser, input: Unit): IO[List[MessageBlockEntry]] =
    val _ = input
    MessageBlockTable.listBlocks(connection, actor.username)
